### ISSUE TITLE
angle_metric vs val_angle_metric

### ISSUE THREAD

**Issue by chilipeppr** [intel-isl/OpenBot#38]
What is the difference between the angle_metric and the val_angle_metric? I ask because I realize the goal is the make the angle_metric as high as possible, approaching 1, but it seems that the val_angle_metric is really the value you want to approach towards 1 and that's the way you guage whether your training is working. What's confusing to me is what's the difference between the two?

When I'm running a training, like I am now, I can see the angle_metric growing (usually) at each epoch, albeit growing slowly, but then at the end of the epoch it calculates the val_angle_metric. I'm just wondering why/how?

![image](https://user-images.githubusercontent.com/7904621/93568302-ca93d600-f944-11ea-93f6-350400784857.png)

Likewise what's the difference with direction_metric and val_direction_metric?

**Comment by thias15** [comment]
There there two datasets, one for training and one for validation/testing. The training dataset should be about 80% of the data and the test dataset about 20%. Now during training, you validate the angle and direction metrics to see how well the network is learning. These are updated at every training step. However, there is a risk that you will overfit to this data. When you actually deploy the robot the images may look slightly different. This is where the test set come in. At the end of every epoch (one run through the training dataset), the current policy is validated against the complete test dataset. This is exactly what the validation metrics show. So, if the training metrics are high, it means the model fits the dataset well. If the validation metrics are high, it means it generalizes to slightly different data. So you want the training metrics as good as possible and the validation metrics as close as possible to them.
